Allow reminder on pointers, reenable packed-structs test

RalfJung opened the pull request.

RalfJung commented:
Oh, and we can also always allow `% 1` as that is always 0. ;)

RalfJung commented:
Fixed.

oli-obk commented on tests/compile-fail/ptr_rem.rs:
```diff
@@ -0,0 +1,5 @@
+fn main() {
+    let val = 13usize;
+    let addr = &val as *const _ as usize;
+    let _ = addr % 2; //~ ERROR access part of a pointer value as raw bytes
```
Why is this problematic? Shouldn't this give us zeroes since `val` is aligned to 4 or 8 bytes so the the remainder is always zero?

RalfJung replied:
Hm I think you are right, I got this the wrong way around.

oli-obk commented on src/operator.rs:
```diff
@@ -193,6 +195,22 @@ impl<'a, 'mir, 'tcx> EvalContextExt<'tcx> for EvalContext<'a, 'mir, 'tcx, super:
             }
 
+            Rem if !signed => {
+                // Doing modulo a multiple of the alignment is allowed
```
shouldn't it be the other way around?

oli-obk replied:
ah no nevermind, I had it swapped in my head

RalfJung commented:
All right, I rebase on top of master and at least locally everything works. (One issue I had got fixed by rebasing, yay :D )

I think I also addressed all of your comments, @oli-obk?

Let's see what CI says...

RalfJung commented:
Oh dang, it still fails with 32bit cross-execution.
